{
	"document": {
		"aggregate_severity": {
			"namespace": "https://access.redhat.com/security/updates/classification/",
			"text": "Moderate"
		},
		"category": "csaf_vex",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright Â© Red Hat, Inc. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "An updated python-sqlalchemy package that fixes one security issue is now\navailable for Red Hat Enterprise Linux 6.\n\nThe Red Hat Security Response Team has rated this update as having moderate\nsecurity impact. A Common Vulnerability Scoring System (CVSS) base score,\nwhich gives a detailed severity rating, is available from the CVE link in\nthe References section.",
				"title": "Topic"
			},
			{
				"category": "general",
				"text": "SQLAlchemy is an Object Relational Mapper (ORM) that provides a flexible,\nhigh-level interface to SQL databases.\n\nIt was discovered that SQLAlchemy did not sanitize values for the limit and\noffset keywords for SQL select statements. If an application using\nSQLAlchemy accepted values for these keywords, and did not filter or\nsanitize them before passing them to SQLAlchemy, it could allow an attacker\nto perform an SQL injection attack against the application. (CVE-2012-0805)\n\nAll users of python-sqlalchemy are advised to upgrade to this updated\npackage, which contains a patch to correct this issue. All running\napplications using SQLAlchemy must be restarted for this update to take\neffect.",
				"title": "Details"
			},
			{
				"category": "legal_disclaimer",
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://access.redhat.com/security/team/contact/",
			"issuing_authority": "Red Hat Product Security is responsible for vulnerability handling across all Red Hat offerings.",
			"name": "Red Hat Product Security",
			"namespace": "https://www.redhat.com"
		},
		"references": [
			{
				"category": "self",
				"summary": "https://access.redhat.com/errata/RHSA-2012:0369",
				"url": "https://access.redhat.com/errata/RHSA-2012:0369"
			},
			{
				"category": "external",
				"summary": "https://access.redhat.com/security/updates/classification/#moderate",
				"url": "https://access.redhat.com/security/updates/classification/#moderate"
			},
			{
				"category": "external",
				"summary": "783305",
				"url": "https://bugzilla.redhat.com/show_bug.cgi?id=783305"
			},
			{
				"category": "self",
				"summary": "Canonical URL",
				"url": "https://access.redhat.com/security/data/csaf/v2/advisories/2012/rhsa-2012_0369.json"
			}
		],
		"title": "Red Hat Security Advisory: python-sqlalchemy security update",
		"tracking": {
			"current_release_date": "2024-08-14T13:08:43+00:00",
			"generator": {
				"date": "2024-08-14T13:08:43+00:00",
				"engine": {
					"name": "Red Hat SDEngine",
					"version": "3.32.0"
				}
			},
			"id": "RHSA-2012:0369",
			"initial_release_date": "2012-03-07T14:24:00+00:00",
			"revision_history": [
				{
					"date": "2012-03-07T14:24:00+00:00",
					"number": "1",
					"summary": "Initial version"
				},
				{
					"date": "2012-03-07T14:25:36+00:00",
					"number": "2",
					"summary": "Last updated version"
				},
				{
					"date": "2024-08-14T13:08:43+00:00",
					"number": "3",
					"summary": "Last generated version"
				}
			],
			"status": "final",
			"version": "3"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_name",
								"name": "Red Hat Enterprise Linux Server (v. 6)",
								"product": {
									"name": "Red Hat Enterprise Linux Server (v. 6)",
									"product_id": "6Server-6.2.z",
									"product_identification_helper": {
										"cpe": "cpe:/o:redhat:enterprise_linux:6::server"
									}
								}
							},
							{
								"category": "product_name",
								"name": "Red Hat Enterprise Linux Workstation (v. 6)",
								"product": {
									"name": "Red Hat Enterprise Linux Workstation (v. 6)",
									"product_id": "6Workstation-6.2.z",
									"product_identification_helper": {
										"cpe": "cpe:/o:redhat:enterprise_linux:6::workstation"
									}
								}
							}
						],
						"category": "product_family",
						"name": "Red Hat Enterprise Linux"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
								"product": {
									"name": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
									"product_id": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-sqlalchemy@0.5.5-3.el6_2?arch=noarch"
									}
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-sqlalchemy-0:0.5.5-3.el6_2.src",
								"product": {
									"name": "python-sqlalchemy-0:0.5.5-3.el6_2.src",
									"product_id": "python-sqlalchemy-0:0.5.5-3.el6_2.src",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-sqlalchemy@0.5.5-3.el6_2?arch=src"
									}
								}
							}
						],
						"category": "architecture",
						"name": "src"
					}
				],
				"category": "vendor",
				"name": "Red Hat"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch as a component of Red Hat Enterprise Linux Server (v. 6)",
					"product_id": "6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch"
				},
				"product_reference": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
				"relates_to_product_reference": "6Server-6.2.z"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-sqlalchemy-0:0.5.5-3.el6_2.src as a component of Red Hat Enterprise Linux Server (v. 6)",
					"product_id": "6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src"
				},
				"product_reference": "python-sqlalchemy-0:0.5.5-3.el6_2.src",
				"relates_to_product_reference": "6Server-6.2.z"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch as a component of Red Hat Enterprise Linux Workstation (v. 6)",
					"product_id": "6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch"
				},
				"product_reference": "python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
				"relates_to_product_reference": "6Workstation-6.2.z"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-sqlalchemy-0:0.5.5-3.el6_2.src as a component of Red Hat Enterprise Linux Workstation (v. 6)",
					"product_id": "6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src"
				},
				"product_reference": "python-sqlalchemy-0:0.5.5-3.el6_2.src",
				"relates_to_product_reference": "6Workstation-6.2.z"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2012-0805",
			"discovery_date": "2012-01-19T00:00:00+00:00",
			"ids": [
				{
					"system_name": "Red Hat Bugzilla ID",
					"text": "783305"
				}
			],
			"notes": [
				{
					"category": "description",
					"text": "Multiple SQL injection vulnerabilities in SQLAlchemy before 0.7.0b4, as used in Keystone, allow remote attackers to execute arbitrary SQL commands via the (1) limit or (2) offset keyword to the select function, or unspecified vectors to the (3) select.limit or (4) select.offset function.",
					"title": "Vulnerability description"
				},
				{
					"category": "summary",
					"text": "python-sqlalchemy: SQL injection flaw due to not checking LIMIT input for correct type",
					"title": "Vulnerability summary"
				},
				{
					"category": "general",
					"text": "The CVSS score(s) listed for this vulnerability do not reflect the associated product's status, and are included for informational purposes to better understand the severity of this vulnerability.",
					"title": "CVSS score applicability"
				}
			],
			"product_status": {
				"fixed": [
					"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
					"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src",
					"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
					"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src"
				]
			},
			"references": [
				{
					"category": "self",
					"summary": "Canonical URL",
					"url": "https://access.redhat.com/security/cve/CVE-2012-0805"
				},
				{
					"category": "external",
					"summary": "RHBZ#783305",
					"url": "https://bugzilla.redhat.com/show_bug.cgi?id=783305"
				},
				{
					"category": "external",
					"summary": "https://www.cve.org/CVERecord?id=CVE-2012-0805",
					"url": "https://www.cve.org/CVERecord?id=CVE-2012-0805"
				},
				{
					"category": "external",
					"summary": "https://nvd.nist.gov/vuln/detail/CVE-2012-0805",
					"url": "https://nvd.nist.gov/vuln/detail/CVE-2012-0805"
				}
			],
			"release_date": "2012-03-07T00:00:00+00:00",
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "Before applying this update, make sure all previously-released errata\nrelevant to your system have been applied.\n\nThis update is available via the Red Hat Network. Details on how to\nuse the Red Hat Network to apply this update are available at\nhttps://access.redhat.com/knowledge/articles/11258",
					"product_ids": [
						"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
						"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src",
						"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
						"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src"
					],
					"restart_required": {
						"category": "none"
					},
					"url": "https://access.redhat.com/errata/RHSA-2012:0369"
				}
			],
			"scores": [
				{
					"cvss_v2": {
						"accessComplexity": "MEDIUM",
						"accessVector": "NETWORK",
						"authentication": "NONE",
						"availabilityImpact": "NONE",
						"baseScore": 4.3,
						"confidentialityImpact": "NONE",
						"integrityImpact": "PARTIAL",
						"vectorString": "AV:N/AC:M/Au:N/C:N/I:P/A:N",
						"version": "2.0"
					},
					"products": [
						"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
						"6Server-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src",
						"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.noarch",
						"6Workstation-6.2.z:python-sqlalchemy-0:0.5.5-3.el6_2.src"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"details": "Moderate"
				}
			],
			"title": "python-sqlalchemy: SQL injection flaw due to not checking LIMIT input for correct type"
		}
	]
}
